### PR TITLE
Trac 10782

### PR DESCRIPTION
**Pull request by jburel**
Few Java integrations test to reflect usage in insight.

### CODE REVIEW

File: components/tools/OmeroJava/test/integration/PojosServiceTest.java [hunk 1]
```diff
@@ -1395,4 +1398,597 @@ public void testLoadContainerHierarchyPlateSpecified()
...
+    			mmFactory.simpleImage(0));
+    	Fileset fileset = new FilesetI();
+    	fileset.addImage(i1);
+    	fileset.addImage(i2);
+    	fileset = (Fileset) iUpdate.saveAndReturnObject(fileset);
+    	//Check that 2 images are linked
+    	assertEquals(fileset.copyImages().size(), 2);
+    	
+    	Parameters param = new ParametersI();
+        Map<String, List<Long>> map = new HashMap<String, List<Long>>(1);
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PojosServiceTest.java]: peculiar indent change here (and other similar code in other methods)

File: components/tools/OmeroJava/test/integration/PojosServiceTest.java [hunk 2]
```diff
@@ -1395,4 +1398,597 @@ public void testLoadContainerHierarchyPlateSpecified()
...
+        map.put(Image.class.getName(), Arrays.asList(i1.getId().getValue()));
+        Map<Long, Map<Boolean, List<Long>>>
+        results = iContainer.getImagesBySplitFilesets(map, param);
+        assertEquals(results.size(), 1);
+        Entry<Long, Map<Boolean, List<Long>>> e;
+        Entry<Boolean, List<Long>> entry;
+        Iterator< Entry<Boolean, List<Long>>> j;
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PojosServiceTest.java]: extra space before `Entry`

**Comment by jburel** [comment]
The formatting of the class is "screwed" up so I am not sure it is worth fixing that in the PR.

**Comment by jburel** [comment]
Forgot to mention:
to test the PR 
`./build.py -f components/tools/OmeroJava/build.xml test -DTEST=integration/PojosServiceTest`

File: components/tools/OmeroJava/test/integration/PojosServiceTest.java [hunk 3]
```diff
@@ -1395,4 +1398,597 @@ public void testLoadContainerHierarchyPlateSpecified()
...
+    		}
+    	}
+    }
+    
+    /**
+     * Test the <code>getImagesBySplitFilesets</code> Image as root.
+     * @throws Exception Thrown if an error occurred.
+     */
+    @Test
+    public void testGetImagesBytSplitFilesetsImageAsRootAll()
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PojosServiceTest.java:1453]: extra `t` in the middle, repeated below

File: components/tools/OmeroJava/test/integration/PojosServiceTest.java [hunk 4]
```diff
@@ -1395,4 +1398,597 @@ public void testLoadContainerHierarchyPlateSpecified()
...
+    	DatasetImageLink link = new DatasetImageLinkI();
+    	link.setParent(d1);
+    	link.setChild((Image) i1.proxy());
+    	iUpdate.saveAndReturnObject(link);
+
+    	link = new DatasetImageLinkI();
+    	link.setParent(d2);
+    	link.setChild((Image) i2.proxy());
+    	iUpdate.saveAndReturnObject(link);
+
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PojosServiceTest.java:1680]: double blank line, and below, but maybe that helps to make the project creation and linking separate and clearer

File: components/tools/OmeroJava/test/integration/PojosServiceTest.java [hunk 5]
```diff
@@ -1395,4 +1398,597 @@ public void testLoadContainerHierarchyPlateSpecified()
...
+    				assertEquals(l.size(), 1);
+    				assertTrue(l.contains(i3.getId().getValue()));
+    			}
+    		}
+    	}
+    }
+    
+    /**
+     * Test the <code>getImagesBySplitFilesets</code> Dataset as root.
+     * One image is not part of a file set.
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PojosServiceTest.java:1861]: thank you, important to include this kind of thing for mixed systems

**Comment by mtbc** [comment]
This is good and useful; I'll merge this later once I'm back working on this branch and actually run the tests.

There's a fair bit of rather similar code repeated. Some might be made briefer with Guava's `ImmutableMap.of` and `ImmutableSet.of` in input arguments but maybe a couple of helper functions for building up test cases and comparing results could be useful. No need for any refactoring, though, the code is fine as it is and we have enough else to keep us busy at present.

**Comment by mtbc** [comment]
`Total tests run: 39, Failures: 0, Skips: 0`

**Pull request merged**